pkg/controllers/unifiedauth/unified_auth_controller.go

```diff
@@ -149,6 +150,9 @@ func (c *Controller) buildImpersonationClusterRole(cluster *clusterv1alpha1.Clus
 }
 
 func (c *Controller) buildImpersonationClusterRoleBinding(cluster *clusterv1alpha1.Cluster) error {
+	if cluster.Spec.ImpersonatorSecretRef == nil {
+		return fmt.Errorf("cluster %s does not have an impersonator secret", cluster.Name)
```

Comment:
If we return an error here, that will lead to an endless retry loop. Is it what we expected?

Reply:
Given the `cluster.Spec.ImpersonatorSecretRef` is an optional field, so it makes sense to unset this field if users don't need this feature, the controller should be able to ignore syncing the RBAC in this situation.

Reply:
This error may directly indicate that the current feature is unavailable. If users don't need this feature, they maybe need to disable the current controller.

Reply:
Is there a possibility that the users only allow some of the clusters can be accessed by aggregated API, and the others can't?

Reply:
It seems reasonable.
Do we need to send a warning here?